rdr_service/genomic/genomic_job_components.py

```diff
@@ -3139,6 +3152,53 @@ def _pull_source_data(self):
         with self.member_dao.session() as session:
             return session.execute(self.manifest_def.source_data).fetchall()
 
+    def _validate_source_data(self, data, manifest_type):
+        invalid = False
```

Comment:
This method works for the immediate need, so no need to change here, but we may want to revisit how this is done in the future. I'm specifically thinking about abstracting the rules, similar to how we validate the files we ingest using the `GenomicFileValidator` class, and the validate_filename() method. That way we can better control the complexity if more files or rules are added.

Reply:
1. Should I add this to the GenomicFileValidator class now, the reason I didn't use it is that seems more catered to files, but I see where we can pass data to it to validate?
2. Tried to make this as general as I could with the idea of it only being targeted towards the AW3s for now as denoted as the priority for this. Plan on extending this later as need be.

Reply:
No, you're right. The AW3s are the priority here and this is needed as soon as possible. I don't think we'd want to use the `GenomicFileValidator`, as you said it is more tailored to file validation rather than data validation. What you have here is good and I've approved it. 
I'm sort of brainstorming that in the future we'd want to validate all output data, and so I am thinking of a new component class. This could be something like `GenomicOutputDataValidator` and a related class for `ValidationRule`. This is a lower priority and will also probably require more fleshed-out requirements, a tech design, etc.